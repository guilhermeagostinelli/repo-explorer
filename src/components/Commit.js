import React from "react";
import styled from "styled-components";
import Card from "./Card";
import H2 from "./H2";
import formatDateTime from "../util/formatDateTime";

// Extends Card styles.
const CommitCard = styled(Card)`
  ::before {
    content: "📝";
  }
`;

const Commit = ({ message, author, comitted_at, html_url }) => (
  <CommitCard>
    <H2 color="#117ed0">{message}</H2>
    <p>Author: {author}</p>
    <p>Committed at {formatDateTime(new Date(comitted_at))}</p>
    <p>
      <a href={html_url} rel="noopener noreferrer" target="_blank">
        See on GitHub
      </a>
    </p>
  </CommitCard>
);

export default Commit;
