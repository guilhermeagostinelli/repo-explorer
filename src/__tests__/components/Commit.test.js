import React from "react";
import { create } from "react-test-renderer";
import Commit from "../../components/Commit";
import "jest-styled-components";

jest.mock("../../components/Card", () => "div");
jest.mock("../../components/H2", () => "h2");
jest.mock("../../util/formatDateTime", () => jest.fn(date => `${date}`));

describe("<Commit />", () => {
  it("should match the snapshot", () => {
    const props = {
      message: "Update README.md",
      author: "Guilherme Agostinelli",
      comitted_at: "2019-08-23T16:20:36Z",
      html_url:
        "https://github.com/guilhermeagostinelli/levenshtein/commit/461fb0ec05e43120e2018bd5a18f461230b99512"
    };

    const commit = create(<Commit {...props} />);
    expect(commit.toJSON()).toMatchSnapshot();
  });
});
