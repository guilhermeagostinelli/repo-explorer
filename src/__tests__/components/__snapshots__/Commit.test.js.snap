// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Commit /> should match the snapshot 1`] = `
.c0::before {
  content: "üìù";
}

<div
  className="c0"
>
  <h2
    color="#117ed0"
  >
    Update README.md
  </h2>
  <p>
    Author: 
    Guilherme Agostinelli
  </p>
  <p>
    Committed at 
    Fri Aug 23 2019 13:20:36 GMT-0300 (Brasilia Standard Time)
  </p>
  <p>
    <a
      href="https://github.com/guilhermeagostinelli/levenshtein/commit/461fb0ec05e43120e2018bd5a18f461230b99512"
      rel="noopener noreferrer"
      target="_blank"
    >
      See on GitHub
    </a>
  </p>
</div>
`;
